src/rgb/contract/conceal.rs

```diff
@@ -22,21 +23,77 @@ pub trait AutoConceal {
     fn conceal_except(&mut self, seals: &Vec<seal::Confidential>) -> usize;
 }
 
-// TODO: Do an auto implementation for and other collection types
-
 impl<T> AutoConceal for Vec<T>
 where
     T: AutoConceal + Ord + Clone,
 {
     fn conceal_except(&mut self, seals: &Vec<seal::Confidential>) -> usize {
         let mut count = 0;
-        let mut new_self = Vec::<T>::new();
+        for item in self.iter_mut() {
+            count += item.conceal_except(seals);
+        }
+        count
```

Comment:
Nit, however more idiomatic implementation for the whole function in a single line

```suggestion
        self.iter_mut().fold(0usize, |sum, &mut item| sum + item_conceal_except(seals))
```

Reply:
Nice!! Used it.